docparse/jsonschema.go

```diff
@@ -240,6 +241,26 @@ start:
 		p.Type = ""
 		name = typ
 
+	// Anonymous struct
+	case *ast.StructType:
+		p.Type = "object"
+		p.Properties = map[string]*Schema{}
+		for _, f := range typ.Fields.List {
+			prop, err := fieldToSchema(prog, "", ref, f)
+			if err != nil {
+				return nil, fmt.Errorf("anon struct: %v", err)
+			}
+
+			propName := f.Names[0].Name
+			if f.Tag != nil {
+				jsonTag := parseRawTag(f.Tag.Value, "json")
```

Comment:
There is the `goutil.Tag` helper for this as well: https://github.com/Teamwork/utils/blob/master/goutil/goutil.go#L215 – I think that does the same?